usecases/security/enforce_security_api_key.go

```diff
@@ -0,0 +1,30 @@
+package security
+
+import (
+	"errors"
+
+	"github.com/checkmarble/marble-backend/models"
+)
+
+type EnforceSecurityApiKeyImpl struct {
+	EnforceSecurity
+	Credentials models.Credentials
+}
+
+func (e *EnforceSecurityApiKeyImpl) CreateApiKey(organizationId string) error {
+	return errors.Join(
+		e.Permission(models.APIKEY_CREATE), e.ReadOrganization(organizationId),
+	)
+}
+
+func (e *EnforceSecurityApiKeyImpl) DeleteApiKey(apiKey models.ApiKey) error {
+	return errors.Join(
+		e.Permission(models.APIKEY_DELETE), e.ReadOrganization(apiKey.OrganizationId),
+	)
+}
+
+func (e *EnforceSecurityApiKeyImpl) ListApiKeys() error {
+	return errors.Join(
+		e.Permission(models.APIKEY_READ),
+	)
+}
```

Comment:
It's not really clear for me :
- sometimes, we check explciitly `e.ReadOrganization(apiKey.OrganizationId)` for read operation too
- sometimes, we do not

Still, usecases on reading info linked to org_id is often based on the orgId from context, but maybe we should do the check explicitly (should be transparent and cost nothing) WDYT ?

Reply:
(see comment in the usecase file)